controllers/util/placement.go

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"reflect"
+
+	"github.com/go-logr/logr"
+	clrapiv1beta1 "github.com/open-cluster-management-io/api/cluster/v1beta1"
+	plrv1 "github.com/stolostron/multicloud-operators-placementrule/pkg/apis/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FinalizePlacement prepares a Placement for scheduling by OCM.
+// NOTE: We Keep the cluster.open-cluster-management.io/experimental-scheduling-disable: "true"
```

Comment:
So if the placement is managed by gitops, then why are even making this change? I assume the user will push changes to the placcment to the gitops in order to disable DR, so why are we making this change? Isn't will overwrite any update to the placement?

Reply:
If the placement is managed by gitops our changes will be overridden, but the annotation will remain, so it should be safe - OCM will not schedule the placement.

Why the user will push change to placement to disable dr? the only change needed is to delete the drpc (which may be managed by gitops as well).

Reply:
> Why the user will push change to placement to disable dr? the only change needed is to delete the drpc (which may be managed by gitops as well).

If the placement is in a Git repository, the user can enable scheduling or disable it through git.

Reply:
It depends on how the placement is deployed - if you apply a placement without the annotation, the annotation is not removed. We don't have any control on the user changes after we finish, we can only leave the placement correct for the current application location.

Reply:
Here’s how I see it: The user requested Ramen to protect an application by setting the scheduler to Ramen or through the annotation. That's essentially all they did and all we're concerned about. Everything else is handled internally by Ramen. Ramen ensures the application is deployed on only one cluster at a time. Once this is done, we leave the placement as it is. We didn't change its spec , label, or annotation; the only thing we added was a finalizer, which we then remove. The rest is the user's job.

Reply:
Yes but we want also to add the annotation when enabling DR see #1442.

I we leave the placement as is, and the user enable OCM scheduling by removing the annotation, and the placement is pointing the wrong cluster (e.g after failover or relocate from the primary cluster), the user will the application data.

This change avoids this by leaving a safe placement that will not change the application location if the user enables OCM scheduling. This is easier to use and safer.